server.go

```diff
@@ -126,14 +127,46 @@ func (server *Server) Stop() {
 		server.Log.Infof("waiting for requests to finish")
 		server.listener.Wait()
 	}
+
+	server.waitForRequests()
+}
+
+// StopDeadline is the hard deadline, after which Server.Stop() does no longer
+// wait for handlers to finish and returns.
+var StopDeadline = time.Minute
+
+// waitForRequests waits until all the handlers from listener and tlsListener
+// are done and close their connections.
+func (server *Server) waitForRequests() {
```

Comment:
name is super confusing. waitForRequests is actually wait for the requests to finish

Function with this name could handle new requests for example ....

Reply:
how about `waitForConnectionsToClose` ?

Reply:
👍